backend/test/controllers/test_usercontroller.py

```diff
@@ -0,0 +1,142 @@
+
+    Returns:
+        UserController: An instance of UserController with a mocked DAO configured to return multiple users.
+    """
+    matched_users = [
+        {'firstName': 'test1_first', 'lastName': 'test1_last', 'email': 'test@test.com'},
+        {'firstName': 'test2_first', 'lastName': 'test2_last', 'email': 'test@test.com'}
+    ]
+    mocked_dao = MagicMock()
+    mocked_dao.find.return_value = matched_users #This configures the find method of the mocked DAO to return a list containing two user objects
+    uc = UserController(dao=mocked_dao)
+    return uc
+
+@pytest.fixture
+def invalid_email():
+    """
+    Fixture for providing an invalid email address string.
+
+    Returns:
+        str: An invalid email address string.
+    """
+    return "test.com"
+
+@pytest.fixture
+def valid_email():
+    """
+    Fixture for providing a valid email address string.
+
+    Returns:
+        str: A valid email address string.
+    """
+    return "test@test.com"
+
+@pytest.fixture
+def other_email():
+    """
+    Fixture for providing another valid email address string.
+
+    Returns:
+        str: Another valid email address string.
+    """
+    return "foo@bar.com"
+
+@pytest.mark.usercontroller
+def test_get_user_by_email_1():
+    """
+    Test case 1: Database fails
+    Email associated with user: Not applicable
+    Multiple Users: Not applicable
+    Email is Valid: Not applicable
+    Database Operation Fails: TRUE
+    Outcome: Exception
+    """
+    mocked_dao = MagicMock()
+    mocked_dao.find.side_effect = Exception() #This configures the find method of the mocked DAO to raise an Exception when called
+    uc = UserController(dao=mocked_dao)
+    # patch the fullmatch method of the regex library
+    with patch('src.controllers.usercontroller.re.fullmatch', autospec=True) as mock_fullmatch: #This patches the re.fullmatch method from the re module used within the UserController class. It ensures that the fullmatch method is replaced with a mock object during the test.
+        mock_fullmatch.return_value = True
+        with pytest.raises(Exception):
+            uc.get_user_by_email({'email': 'email'}) #This context manager asserts that the code block within it raises an Exception. If the expected exception is not raised, the test will fail.
+
+@pytest.mark.usercontroller
+def test_get_user_by_email_2(sut_many, capfd, valid_email):
+    """
+    Test case 2: Valid email associated to multiple users
+    Email  associated with user: TRUE
+    Multiple Users: TRUE
+    Email is Valid: TRUE
+    Database Operation Fails: FALSE
+    Outcome: User object (first in list); Print warning
+    """
+    # return the first one and print a warning message containing the email address
+    sut_many.get_user_by_email(valid_email)
+    # Check if the warning message is printed
+    out, err = capfd.readouterr()
+    assert valid_email in out
+
+@pytest.mark.usercontroller
+def test_get_user_by_email_3(sut_one, invalid_email):
+    """
```

Comment:
we should assert that we get 1) correct user[0] user and 2) print error
we should have two separate assertions
now our test case asserts only print message
by the test cases docstring states that both should be the outcome